### PR TITLE
Support for HLS transcription (resolves #62)

### LINKED ISSUE #62: Transcribing a HLS stream?
I've tried transcribing an HLS stream by inputing a link to a .m3u8 file, but it does not seem to work. Is that feature supported somehow?

### PR DESCRIPTION
**Pull request by jhormigo**
This resolves https://github.com/collabora/WhisperLive/issues/62 feature requets.

It can be tested with:

```
client = TranscriptionClient(host, port, is_multilingual=True, lang="en", translate=False) 
client(hls_url="http://as-hls-ww-live.akamaized.net/pool_904/live/ww/bbc_1xtra/bbc_1xtra.isml/bbc_1xtra-audio%3d96000.norewind.m3u8") 
```
This could be added to the README.me

### CODE REVIEW

**Comment by jhormigo** [comment]
This resolves #62  feature requets.

It can be tested with: 

`client = TranscriptionClient(host, port, is_multilingual=True, lang="en", translate=False)
client(hls_url="http://as-hls-ww-live.akamaized.net/pool_904/live/ww/bbc_1xtra/bbc_1xtra.isml/bbc_1xtra-audio%3d96000.norewind.m3u8")
`

This could be added to the README.me

File: whisper_live/client.py [hunk 1]
```diff
@@ -344,6 +345,46 @@ def write_audio_frames_to_file(self, frames, file_name):
...
+            hls_url (str): The URL of the HLS stream source.
+        """
+        print("[INFO]: Connecting to HLS stream...")
+        process = None  # Initialize process to None
+
+
+        try:
+            # Launch an FFMPEG process to connect to the HLS stream
+            command = [
+                'ffmpeg',
```
Comment (zoq) [inline whisper_live/client.py:362]: With this in place, we also should mention in the README that ffmpeg is a requirment.

**Review by zoq — COMMENTED** [review]
Thanks, going to test it out tomorrow.

File: whisper_live/client.py (hunk 1 above)
Comment (jhormigo) [reply whisper_live/client.py]: I've now changed the requirement of having ffmpeg installed by using the `ffmpeg-python` library. I've now resubmitted a new version of the code. Please check.

File: README.md [hunk 2]
```diff
@@ -41,6 +41,12 @@ Unlike traditional speech recognition systems that rely on continuous audio stre
     ```
     This command captures audio from the microphone and sends it to the server for transcription. It uses the same options as the previous command, enabling the multilingual feature and specifying the target language and task.
 
+    - To trasncribe from a HLS stream:
+    ```python
+      client = TranscriptionClient(host, port, is_multilingual=True, lang="en", translate=False) 
+      client(hls_url="http://domain.url/playlist.m3u8")
```
Comment (makaveli10) [inline README.md:47]: Is this a test url? because when I test this out; this has no transcript at all.

Comment (jhormigo) [reply README.md:47]: @makaveli10 the test link is in the first comment [here](https://github.com/collabora/WhisperLive/pull/73#issue-2034040337).

What you referenced is a non FQDN sample url for the README.md purposes.

Comment (jhormigo) [reply README.md:47]: Given that you're not providing a file in this example on the README.md

  ```
from whisper_live.client import TranscriptionClient
  client = TranscriptionClient("localhost", 9090, is_multilingual=True, lang="hi", translate=True)
  client(audio_file_path)
```

We should keep consistency for users on the same format.

Comment (jhormigo) [reply README.md:47]: @zoq Are we adding then some samples into the README.md or leaving it as it is? Let me know.

Comment (makaveli10) [reply README.md:47]: @jhormigo thanks for pointing that out. I'll add a sample to the repo and mention to the readme. Meanwhile, can you please do the same for HLS stream as well?

Comment (jhormigo) [reply README.md]: @makaveli10 I've added it now. You can probably merge this now.

**Review by makaveli10 — APPROVED** [review]

**Comment by makaveli10** [comment]
Thanks @jhormigo for putting this together.

**Pull request merged**